### PR TITLE
fix: describe distribute table

### LINKED ISSUE #923: column constrain `PRIMARY KEY` doesn't take efforts when placed after column's definition
### What type of bug is this?

Incorrect result

### What subsystems are affected?

Standalone mode, Frontend, Datanode

### What happened?

The result of `host STRING PRIMARY KEY`
```sql
mysql> CREATE TABLE demo (timestamp BIGINT TIME INDEX, host STRING PRIMARY KEY, value DOUBLE);
mysql> desc table demo;
+-----------+---------+------+---------+---------------+
| Field     | Type    | Null | Default | Semantic Type |
+-----------+---------+------+---------+---------------+
| timestamp | Int64   | NO   |         | TIME INDEX    |
| host      | String  | YES  |         | VALUE         |
| value     | Float64 | YES  |         | VALUE         |
+-----------+---------+------+---------+---------------+
3 rows in set (0.01 sec)
```

It should work like this:
```sql
mysql> CREATE TABLE demo (timestamp BIGINT TIME INDEX, host STRING, value DOUBLE, PRIMARY KEY(host));
mysql> desc table demo;
+-----------+---------+------+---------+---------------+
| Field     | Type    | Null | Default | Semantic Type |
+-----------+---------+------+---------+---------------+
| timestamp | Int64   | NO   |         | TIME INDEX    |
| host      | String  | YES  |         | PRIMARY KEY   |
| value     | Float64 | YES  |         | VALUE         |
+-----------+---------+------+---------+---------------+
3 rows in set (0.01 sec)
```

### What operating system did you use?

macOS 12.3 21E230 arm64

### Relevant log output and stack trace

_No response_

### How can we reproduce the bug?

Using the following sqls

```sql
CREATE TABLE demo (timestamp BIGINT TIME INDEX, host STRING PRIMARY KEY, value DOUBLE);
```

```sql
DESC TABLE demo;
```

### PR DESCRIPTION
**Pull request by MichaelScofield**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

Fix a bug that `DESC table` is not working correctly for distribute table when primary keys are defined in column options.

Also:

- refactor some error msgs to make them more clear
- unify sqlness test case for creating table

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [x]  I have added the necessary unit tests and integration tests.

## Refer to a related PR or issue link (optional)

close #923

### CODE REVIEW

File: src/frontend/src/expr_factory.rs [hunk 1]
```diff
@@ -111,6 +136,17 @@ fn find_primary_keys(constraints: &[TableConstraint]) -> Result<Vec<String>> {
...
+    ensure!(
+        columns_pk.is_empty() || constraints_pk.is_empty(),
+        IllegalPrimaryKeysDefSnafu {
+            msg: "found definitions of primary keys in multiple places"
+        }
+    );
+
+    let mut primary_keys = Vec::with_capacity(columns_pk.len() + constraints_pk.len());
+    primary_keys.append(columns_pk);
+    primary_keys.append(constraints_pk);
```
Comment (waynexia) [inline src/frontend/src/expr_factory.rs:149]: use [extend](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.extend_from_slice) can reduce one allocation

File: tests/cases/standalone/order/order_variable_size_payload.sql [hunk 2]
```diff
@@ -123,3 +123,5 @@ DROP table test7;
 DROP table test8;
 
 DROP TABLE DirectReports;
+
+-- TODO(LFC): Seems creating distributed table has some column schema related issues, look into "order_variable_size_payload" test cases.
```
Comment (waynexia) [inline tests/cases/standalone/order/order_variable_size_payload.sql:127]:
```suggestion
-- TODO(LFC): Seems creating distributed table has some column schema related issues, look into "order_variable_size_payload" test case.
```

**Review by waynexia — COMMENTED** [review]
Is this bug only occur in a distributed table? Looks like `find_primary_keys()` also affects other modes.

**Comment by MichaelScofield** [comment]
`find_primary_keys` is only called in distributed table creation code path, so this bug only exists in distribute mode. Other comments are resolved, PTAL @waynexia

**Review by waynexia — APPROVED** [review]

**Review by sunng87 — APPROVED** [review]

**Pull request merged**